pkg/apis/v1beta1/nodepool.go

```diff
@@ -124,7 +124,7 @@ type Budget struct {
 	// This is required if Schedule is set.
 	// This regex has an optional 0s at the end since the duration.String() always adds
 	// a 0s at the end.
-	// +kubebuilder:validation:Pattern=`^([0-9]+(m|h)+(0s)?)$`
+	// +kubebuilder:validation:Pattern=`^((([0-9]+(h|m))|([0-9]+h[0-9]+m))(0s)?)$`
```

Comment:
Devils advocate: can't we just lower the unit we are using? 1h10m is just 70m. Maybe it gets more complex with 12d10m vs 730m. 

I commonly see 
```
  consolidateAfter: 300s
  consolidationPolicy: WhenEmpty
  expireAfter: 720h
```

If we do move to a 1h10m pattern, we should probably do it everywhere to be consistent. [here](https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1beta1/nodepool.go#L85) for example we follow a different pattern. It will be fairly confusing for users to see they can set 1h10m in one place but not everywhere

Reply:
This PR would allow both to happen. We currently promise both format to customer and this is addressing a bug

Reply:
It seems in your link @Bryce-Soghigian that the only difference in these two regexes is the order. Seems compound durations are enabled there as well.